test_cycles.py

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/python3
```

Comment:
In my opinion, this file isn't necessary. If the script can find the cycle in the test file above then it can obviously find them in the constructed graph below.